### PR TITLE
Support strings in events by calling .clone()

### LINKED ISSUE #61: Error when using string in events
Currently there's a Rust compiler error if an event includes a String field. See eg: https://beta.solpg.io/6380b97677ea7f12846aef44

Error is:

```sh
Building...
   error[E0507]: cannot move out of dereference of `Ref<'_, MyEvent>`
 --> /src/dot/program.rs:1:586
  |
1 | ... { title : self . borrow () . title }) } } impl MyEvent { pub fn __new__ (title : String) -> Mutable < Self > { let obj = Mutable :: n...
  |               ^^^^^^^^^^^^^^^^^^^^^^^^ move occurs because value has type `String`, which does not implement the `Copy` trait
```

### PR DESCRIPTION
**Pull request by mcintyre94**
Since strings in Rust don't implement `Copy`, we need to call `.clone()` when we read these fields of an event. This PR adds logic to add `.clone()` for only strings.

I've also added an event example which shows various different field types in events. I found that lists and arrays also don't work, but will also need further work, so I've opened a separate issue for them: #64 

Closes #61

### CODE REVIEW

**Comment by ameliatastic** [comment]
Definitely a symptom of a deeper problem with strings. I started using strings in init seeds pretty recklessly (essentially letting string literals turn into Rust string literals because it was easier). I'll look pretty closely at this, maybe even adding some unrelated extra fixes if that's okay with you.

**Comment by ameliatastic** [comment]
Here's an optional second way of doing things, which uses the original type information instead of intuiting based on the generated type's name. Might make things easier in the future, thoughts?

File: src/core/compile/build/mod.rs [hunk 1]
```diff
@@ -1086,7 +1086,7 @@ impl TryFrom<CheckOutput> for BuildOutput {
                                                 } else {
                                                     TypeDef::Struct(Struct {
                                                         name,
-                                                        fields: fields.into_iter().map(|(name, ty, _)| (name, ty)).collect(),
+                                                        fields: fields.into_iter().map(|(name, ty_expr, ty)| (name, ty_expr, ty)).collect(),
```
Comment (mcintyre94) [inline src/core/compile/build/mod.rs:1089]: Can we drop this iter/map now?

**Review by mcintyre94 — COMMENTED** [review]
Awesome that's way better! Should work for other data types too without us needing to keep another match up to date so definitely prefer that.

File: src/core/compile/build/mod.rs (hunk 1 above)
Comment (ameliatastic) [reply src/core/compile/build/mod.rs:1089]: lol yeah good catch

**Comment by mcintyre94** [comment]
Awesome this LGTM now! TY for the improvement!

**Comment by ameliatastic** [comment]
Cool! I'll just merge it, I was looking to see if there was any extra problems with strings that could be solved here but it ended up being a totally separate issue.

**Pull request merged**